tensorflow/contrib/learn/python/learn/datasets/mnist.py

```diff
@@ -223,23 +223,25 @@ def fake():
   TEST_IMAGES = 't10k-images-idx3-ubyte.gz'
   TEST_LABELS = 't10k-labels-idx1-ubyte.gz'
 
-  local_file = base.maybe_download(TRAIN_IMAGES, train_dir,
-                                   SOURCE_URL + TRAIN_IMAGES)
+  def download_local_file(fname):
+    try:
+      return base.maybe_download(fname, train_dir, SOURCE_URL + fname)
```

Comment:
We could make maybe_download take a filename, a directory, and either a URL or a list of URLs. if it gets a string (single url) behavior is same as present. if it gets an array of strings, then it tries them in sequence until successful. that way we can generalize the concept of multiple sources across the codebase in a backwards compatible way.